doc/source/whatsnew/v0.18.1.txt

```diff
@@ -76,8 +76,7 @@ API changes
 
 
 
-
-
+- In ``read_csv``, the Python parser now raises an ``EmptyDataError`` in response to empty columns or header (:issue:`12506`)
```

Comment:
You can easily find this empty `StopIteration` as follows:
~~~python
>>> from pandas.compat import StringIO
>>> from pandas import read_csv
>>> data = ""
>>> read_csv(StringIO(data), engine='c')
...
ValueError: No columns to parse from file
>>> read_csv(StringIO(data), engine='python') # master
...
StopIteration
~~~
Using my PR:
>>> read_csv(StringIO(data), engine='python') # master
...
pandas.io.parsers.EmptyDataError: No columns to parse from file
~~~